app/pkg/build.gradle

```diff
@@ -109,8 +109,25 @@ jlink {
         }
         if (OperatingSystem.current().isMacOsX()) {
             imageOptions = [
-                    '--icon', "${projectDir}/src/main/icons/Icon.icns"
+                    '--icon', "${projectDir}/src/main/icons/Icon.icns",
+                    '--mac-sign',
+                    // jpackage is expecting to use a "Developer ID Application" certificate for
+                    // signing the app, so make sure your keychain contains this certificate, along
+                    // with the corresponding private key, if you wish to sign.  The certificate
+                    // should have a full name of:
+                    // 'Developer ID Application: The International Data Organization For Transport (BF2U75HN4D)'
+                    // jpackage will supply the 'Developer ID Application: ' prefix when calling the
+                    // codesign tool.
+                    '--mac-signing-key-user-name',
+                    'The International Data Organization For Transport (BF2U75HN4D)',
+                    // Entitlements are required for notarization of the app.
+                    '--mac-entitlements',
+                    "${projectDir}/src/main/mac-resources/entitlements.plist",
```

Comment:
jpackage has support for codesigning the app, so I went with that option initially.  However, there is maybe an argument we should remove this change and make an explicit codesigning request in the `package_installers.yml` workflow instead.  That way, devs could still build the app locally without having to worry about codesigning setup.  Thoughts?

Reply:
@bdferris-v2 I'll let @maximearmstrong approve this, if he has time. If not, I'll do it tomorrow before our meeting.

Reply:
@fredericsimard not critical that we have this checked in beforehand.  We should be able to test it either way.

@maximearmstrong I took a slightly different approach here than I was originally thinking.  It turns out that the number of codesign commands run by jpackage is more than I initially thought.  As such, I stuck with the approach of letting jpackage sign the app but I wrapped the flags in a Property check for `sign-app`, which is disabled by default.  If someone needs to sign the app, including the workflow, they just have to add `-Psign-app=true` in the call to gradle.